### PR TITLE
Add PolBlogs Dataset

### PR DESCRIPTION
**Pull request by soham-chitnis10**
Adding PolBlogs dataset 
Torch_geometric implementation [PolBlogs](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/datasets/polblogs.html#PolBlogs )

### CODE REVIEW

File: src/PolBlogs/PolBlogs.jl [hunk 1]
```diff
@@ -0,0 +1,83 @@
+export PolBlogs
+
+
+"""
+The Political Blogs dataset from the `"The Political Blogosphere and
+the 2004 US Election: Divided they Blog" <https://dl.acm.org/doi/10.1145/1134271.1134277>`_ paper.
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:6]:
```suggestion
    PolBlogs
 
The Political Blogs dataset from the [The Political Blogosphere and
the 2004 US Election: Divided they Blog](https://dl.acm.org/doi/10.1145/1134271.1134277) paper.
```

File: src/PolBlogs/PolBlogs.jl [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+`Polblogs` is a graph with 1,490 vertices (representing political blogs) and 19,025 edges (links between blogs).
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:8]:
```suggestion
`PolBlogs` is a graph with 1,490 vertices (representing political blogs) and 19,025 edges (links between blogs).
```

File: src/PolBlogs/PolBlogs.jl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+The links are automatically extracted from a crawl of the front page of the blog. 
+
+Each vertex receives a label indicating the political leaning of the blog: liberal or conservative.
+# Interface
+- [`PolBlogs.adjacency`](@ref)
+ - [`PolBlogs.labels`](@ref)
+# Utilities
+- [`PolBlogs.download`](@ref)
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:17]:
```suggestion
Each vertex receives a label indicating the political leaning of the blog: liberal or conservative.

# Interface

- [`PolBlogs.adjacency`](@ref)
- [`PolBlogs.labels`](@ref)
```
Let's not expose download, it's not much useful

File: src/PolBlogs/PolBlogs.jl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Trigger the (interactive) download of the full dataset into
+"`dir`". If no `dir` is provided the dataset will be
+downloaded into "~/.julia/datadeps/$DEPNAME".
+This function will display an interactive dialog unless
+either the keyword parameter `i_accept_the_terms_of_use` or
+the environment variable `DATADEPS_ALWAY_ACCEPT` is set to
+`true`. Note that using the data responsibly and respecting
+copyright/terms-of-use remains your responsibility.
+"""
+download(args...;kw...) = download_dep(DEPNAME,args...;kw...)
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:42]:
```suggestion
```

File: src/PolBlogs/PolBlogs.jl [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    register(DataDep(DEPNAME,
+    """
+    Dataset : The $DEPNAME dataset
+    Website : $LINK
+    """,
+    LINK,
+    post_fetch_method = unpack
+    ))
+end
+"""
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:54]:
```suggestion
end

"""
```

File: src/PolBlogs/PolBlogs.jl [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    adjacency(; dir = nothing)
+
+Returns a 19025 x 2 matrix containing edge indices
+
+```juliarepl
+using MLDatasets : PolBlogs
+adj = PolBlogs.adjacency()
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:61]:
```suggestion
```julia-repl
using MLDatasets : PolBlogs

adj = PolBlogs.adjacency()
```

File: src/PolBlogs/PolBlogs.jl [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+# Utilities
+- [`PolBlogs.download`](@ref)
+"""
+module PolBlogs
+
+using DataDeps
+using DelimitedFiles
+using ..MLDatasets: bytes_to_type, datafile, download_dep, download_docstring
+
+export download,adjacency,labels
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:25]:
```suggestion
export adjacency, labels
```

File: src/PolBlogs/PolBlogs.jl [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+const LINK = "https://netset.telecom-paris.fr/datasets/polblogs.tar.gz"
+const DEPNAME = "PolBlogs"
+const DATA = ["adjacency.csv","labels.csv"]
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:29]:
```suggestion
const DATA = ["adjacency.csv", "labels.csv"]
```

File: src/PolBlogs/PolBlogs.jl [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    Website : $LINK
+    """,
+    LINK,
+    post_fetch_method = unpack
+    ))
+end
+"""
+    adjacency(; dir = nothing)
+
+Returns a 19025 x 2 matrix containing edge indices
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:57]: The `adjacency` name is confusing, one would think of a `N x N` adjacency matrix. 
Let's rename this function to `edge_index` and explain that the 2 columns contain the target nodes and the source nodes of each edge.

File: src/PolBlogs/PolBlogs.jl [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+```juliarepl
+using MLDatasets : PolBlogs
+adj = PolBlogs.adjacency()
+```
+"""
+function adjacency(;dir=nothing)
+    path = datafile(DEPNAME,DATA[1],dir)
+    adj = readdlm(path,',')
+    Matrix{Int64}(adj)
+end
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:68]:
```suggestion
"""

function adjacency(; dir = nothing)
    path = datafile(DEPNAME, DATA[1], dir)
    adj = readdlm(path, ',')
    return Matrix{Int64}(adj)
end
```
please use blanks as for the style's of most julia repos

File: src/PolBlogs/PolBlogs.jl [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+using MLDatasets : PolBlogs
+labels = Polblogs.labels()
+```
+"""
+function labels(;dir=nothing)
+    path = datafile(DEPNAME,DATA[2],dir)
+    labels = readdlm(path,',')
+    Matrix{Int64}(labels)
+end
+end #module
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:83]:
```suggestion
function labels(; dir=nothing)
    path = datafile(DEPNAME, DATA[2], dir)
    labels = readdlm(path, ',')
    Matrix{Int64}(labels)
end

end #module

```

**Comment by soham-chitnis10** [comment]
@CarloLucibello Before making the changes, It was passing documentation tests but for some reason, after I changed the name of the function it keeps saying no doc reference found for PolBlogs.edge_index whereas I already have provided docs could you look into this issue?

File: src/PolBlogs/PolBlogs.jl [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+    edge_index(; dir = nothing)
+
+Returns a 19025 x 2 matrix containing edge indices
+
+```juliarepl
+using MLDatasets : PolBlogs
+adj = PolBlogs.edge_index()
+```
+"""
+
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:54]: No newline between a function and its docstring

File: src/PolBlogs/PolBlogs.jl [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+function edge_index(; dir = nothing)
+    path = datafile(DEPNAME, DATA[1], dir)
+    adj = readdlm(path, ',')
+    return Matrix{Int64}(adj)
+end
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:59]: Newline here

File: src/PolBlogs/PolBlogs.jl [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+"""
+    labels(; dir = nothing)
+
+Returns a matrix of labels of size 1490 x 1
+
+```juliarepl
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:65]:
```suggestion
```julia-repl
```

File: src/PolBlogs/PolBlogs.jl [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    post_fetch_method = unpack
+    ))
+end
+
+"""
+    edge_index(; dir = nothing)
+
+Returns a 19025 x 2 matrix containing edge indices
+
+```juliarepl
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:49]:
```suggestion
```julia-repl
```

File: src/PolBlogs/PolBlogs.jl [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+    """,
+    LINK,
+    post_fetch_method = unpack
+    ))
+end
+
+"""
+    edge_index(; dir = nothing)
+
+Returns a 19025 x 2 matrix containing edge indices
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:47]: Please expand the docstrings as I suggested

File: src/PolBlogs/PolBlogs.jl [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+
+```juliarepl
+using MLDatasets : PolBlogs
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:50]:
```suggestion
using MLDatasets: PolBlogs
```

File: src/PolBlogs/PolBlogs.jl [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+
+```julia-repl
+using MLDatasets: PolBlogs
+labels = PolBlogs.labels()
+```
+"""
+function labels(; dir=nothing)
+    path = datafile(DEPNAME, DATA[2], dir)
+    labels = readdlm(path, ',')
+    Matrix{Int64}(labels)
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:73]:
```suggestion
    Matrix{Int64}(labels) |> vec
```

File: src/PolBlogs/PolBlogs.jl [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+function edge_index(; dir = nothing)
+    path = datafile(DEPNAME, DATA[1], dir)
+    adj = readdlm(path, ',')
+    return Matrix{Int64}(adj)
+end
+
+"""
+    labels(; dir = nothing)
+
+Returns a matrix of labels of size 1490 x 1
```
Comment (CarloLucibello) [inline src/PolBlogs/PolBlogs.jl:63]:
```suggestion
Returns a vector containing the 1490 labels.
```

File: test/tst_polblogs.jl [hunk 20]
```diff
@@ -0,0 +1,12 @@
+data_dir = withenv("DATADEPS_ALWAY_ACCEPT"=>"true") do
+    datadep"PolBlogs"
+end
+
+@testset "PolBlogs" begin
+    adj = PolBlogs.edge_index()
+    labels = PolBlogs.labels()
+    @test adj isa Matrix{Int64}
+    @test labels isa Matrix{Int64}
```
Comment (CarloLucibello) [inline test/tst_polblogs.jl:9]:
```suggestion
    @test labels isa Vector{Int64}
```

File: test/tst_polblogs.jl [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+    @test size(adj) == (19025,2)
+    @test size(labels) == (1490,1)
```
Comment (CarloLucibello) [inline test/tst_polblogs.jl:11]:
```suggestion
    @test size(labels) == (1490,)
```

**Review by CarloLucibello — APPROVED** [review]

**Comment by CarloLucibello** [comment]
Thanks!

**Pull request merged**